src/hal/alarm.cpp

```diff
@@ -0,0 +1,76 @@
+
+#include <stdbool.h>
+
+#include "../hal/hal.h"
+#include "../hal/alarm.h"
+#include "../hal/timer.h"
+
+#include <Arduino.h>
+
+int alarmHalInit(void)
+{
+  pinMode(ALARM_PIN, OUTPUT);
+}
+
+int alarmHalRing(unsigned int pattern)
+{
+  static unsigned int currentPattern = ALARM_HAL_OFF;
+  static bool alarmOn = false;
+  static struct timer timer;
+  uint32_t alarmTime;
+
+  // Check to see if new tone needs to be played
+  if (pattern != currentPattern) {
+    noTone(ALARM_PIN);
+    currentPattern = pattern;
+    alarmOn = false;
+  }
+  
+  // Convert alarm frequency into time in seconds
+  switch (pattern) {
+    case ALARM_HAL_0_25HZ:
+    alarmTime = 4 SEC;
+    break;
+    case ALARM_HAL_0_5HZ:
+    alarmTime = 2 SEC;
+    break;
+    case ALARM_HAL_1HZ:
+    alarmTime = 1 SEC;
+    break;
+    default:
+    alarmTime = 0 SEC;
+    break;
+  }
+  
+  if (alarmOn) {
+    // If a blinking buzzer, check the running timer
+    if ((currentPattern == ALARM_HAL_0_25HZ) ||
```

Comment:
Can we replace this if statement with one that just checks that `currentPattern` isn't `ALARM_HAL_OFF` or `ALARM_HAL_CONSTANT`?

Reply:
or just `ALARM_HAL_CONSTANT` because looking through the code, it seems it will only hit this `currentPattern` is set to `ALARM_HAL_CONSTANT`.

Reply:
Indeed we can

Reply:
Done in latest commits